modules/create_timepoint/jsx/create_timepoint.js

```diff
@@ -0,0 +1,352 @@
+      state.form.value.subproject
+    ])[0];
+    state.form.display.visit = true;
+    this.setState(state);
+  }
+  /**
+   * Set the form data based on state values of child elements/components
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+  setForm(formElement, value) {
+    const state = Object.assign({}, this.state);
+    state.form.value[formElement] = value;
+    this.setState(state);
+    if (formElement === 'subproject') {
+      this.handleVisitLabel();
+    }
+  }
+
+  /**
+   * Populate the elements of errors to display.
+   *
+   * @param {object} values - for individual form element.
+   */
+  populateErrors(values) {
+    let errors = [];
+    Object.keys(values).forEach(function(key) {
+      errors.push(
+        <div className='col-xs-12 col-sm-12 col-md-12'>
+          <label className='error form-group'>
+            {values[key]}
+          </label>
+        </div>
+      );
+    });
+    this.setState({errors: errors});
+  }
+
+  /**
+   * Handle form submission
+   * @param {object} e - Form submission event
+   */
+  handleSubmit(e) {
+    const state = Object.assign({}, this.state);
+    const send = this.urlSearchParams({
+      command: 'create',
+      candID: state.url.params.candID,
+      identifier: state.url.params.identifier,
+      subproject: state.form.value.subproject,
+      psc: state.form.value.psc,
+      visit: state.form.value.visit,
+    });
+    const url = this.props.dataURL + '/create_timepoint/AjaxTimepoint';
+    fetch(
+      url, {
+        method: 'POST',
+        mode: 'same-origin',
+        credentials: 'include',
+        redirect: 'follow',
+        headers: {
+          'Accept': 'application/json, text/plain, */*',
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: send,
+      }
+    ).then((response) => response.json())
+      .then(
+        (data) => {
+          if (data.status === 'error') {
+            // Populate the form errors.
+            if (data.errors) {
+              this.populateErrors(data.errors);
+            }
+          } else {
+            this.setState({success: true});
+          }
+    });
+  }
+  /**
+   * @return {DOMRect}
+   */
+  render() {
+    // Waiting for async data to load.
+    if (!this.state.isLoaded) {
+      return <Loader/>;
+    }
+    // Include form errors.
+    const errors = this.state.errors;
+    // Include subproject select.
+    const subproject = this.state.form.display.subproject ? (
+      <SelectElement
+        id={'subproject'}
+        name={'subproject'}
+        label={'Subproject'}
+        value={this.state.form.value.subproject}
+        options={this.state.form.options.subproject}
+        onUserInput={this.setForm}
+        emptyOption={false}
+        disabled={false}
+        required={true}
+      />
+    ) : '';
```

Comment:
would it maybe be better to set as null where the empty string is?

Reply:
@zaliqarosli I've never used null in this situation here. I've seen empty string used elsewhere online when looking at react code and so I sort of think empty string is preferred. Thoughts?

Reply:
apparently we should be using false, null, or undefined https://stackoverflow.com/questions/30097091/correct-way-to-define-an-empty-dom-element-in-react